### PR TITLE
[GH-1237] api: Handling maxKeys=0 case with a empty response

### LINKED ISSUE #1237: api: List Object Test failure when maxKeys set to 0
When MaxKeys set to 0 in the test case here https://github.com/hackintoshrao/minio/blob/a5c8fe3a8b70e364a35a0829d59fcfb8b2a832b4/pkg/fs/fs-bucket-listobjects_test.go#L275 , the test fails since the behavior deviates from S3. 

S3 returns an empty response, but Minio server sets the value to `listObjectsLimit` (1000 ) .  A simple check has to be done here https://github.com/minio/minio/blob/master/pkg/fs/fs-bucket-listobjects.go#L67 .

### PR DESCRIPTION
**Pull request by hackintoshrao**
Fixes #1237 .

### CODE REVIEW

File: pkg/fs/fs-bucket-listobjects.go [hunk 1]
```diff
@@ -64,7 +64,12 @@ func (fs Filesystem) ListObjects(bucket, prefix, marker, delimiter string, maxKe
 		}
 	}
 
-	if maxKeys <= 0 || maxKeys > listObjectsLimit {
+	// Return empty resposne for a valid request when maxKeys is 0.
```
Comment (harshavardhana) [inline pkg/fs/fs-bucket-listobjects.go]: s/resposne/response

File: pkg/fs/fs-bucket-listobjects.go [hunk 2]
```diff
@@ -64,7 +64,12 @@ func (fs Filesystem) ListObjects(bucket, prefix, marker, delimiter string, maxKe
...
+	if maxKeys == 0 {
+		return result, nil
+	}
+
+	if maxKeys < 0 || maxKeys > listObjectsLimit {
 		maxKeys = listObjectsLimit
 	}
```
Comment (harshavardhana) [inline pkg/fs/fs-bucket-listobjects.go]: Looks like test is missing, can you update that as well?

Comment (hackintoshrao) [inline pkg/fs/fs-bucket-listobjects.go]: You want me to push both in same update ?

Comment (hackintoshrao) [inline pkg/fs/fs-bucket-listobjects.go]: If so then close this PR, I already have tests in another branch, I'll send the fix with the tests.

Comment (abperiasamy) [inline pkg/fs/fs-bucket-listobjects.go]: If you already have tests in another PR its is OK. Your case is an exception, since you own the tests.

**Pull request merged**